ignorePackageGlobs doesn't appear to be working

Hey, great tool! This has been really useful to track down places I can apply some new error context packages I've been working on to provide additional structured information to errors.

After applying error contexts across our project I noticed a few places where it was unnecessary so I decided to disable reporting of imports from local packages (/pkg, etc)

However, maybe I'm misunderstanding this config option, but it doesn't appear to do what I expected.

Here's the configuration file: https://github.com/Southclaws/storyden/blob/main/.golangci.yml#L28

And here are the offending lines: 


- https://github.com/Southclaws/storyden/blob/main/pkg/transports/http/bindings/threads.go#L39
- https://github.com/Southclaws/storyden/blob/main/pkg/transports/http/bindings/threads.go#L52

I assumed that a `ignorePackageGlobs` value of `github.com/Southclaws/storyden/*` would ignore these two lines as they are functions that are imported from within this pattern:

- `i.thread_svc.Create`: `github.com/Southclaws/storyden/pkg/services/thread.Service.Create`
- `i.thread_svc.ListAll`: `github.com/Southclaws/storyden/pkg/services/thread.Service.Create`

The actual output from running `golangci-lint run`:

```
storyden on  main [!] via 🐹 v1.18.3 on ☁️  (eu-central-1) 
❯ golangci-lint run

pkg/transports/http/bindings/threads.go:39:10: error returned from interface method should be wrapped: sig: func (github.com/Southclaws/storyden/pkg/services/thread.Service).Create(ctx context.Context, title string, body string, authorID github.com/Southclaws/storyden/pkg/resources/account.AccountID, categoryID github.com/Southclaws/storyden/pkg/resources/category.CategoryID, tags []string) (*github.com/Southclaws/storyden/pkg/resources/thread.Thread, error) (wrapcheck)
                return err
                       ^
pkg/transports/http/bindings/threads.go:52:10: error returned from interface method should be wrapped: sig: func (github.com/Southclaws/storyden/pkg/services/thread.Service).ListAll(ctx context.Context, before time.Time, max int) ([]*github.com/Southclaws/storyden/pkg/resources/thread.Thread, error) (wrapcheck)
                return err
                       ^
```

I wondered if it was golangci lint just using an outdated version but this occurs on the latest standalone version of wrapcheck too.

I tried with a .wrapcheck.yaml file with:

```yaml
ignorePackageGlobs:
  - github.com/Southclaws/storyden/*
  - github.com/Southclaws/storyden/pkg/*
  - pkg/*
  - ./pkg/*
```

but I can't seem to get any patterns to ignore correctly.

Thanks!

--- a/wrapcheck/testdata/config_ignorePackageGlobs_interface/.wrapcheck.yaml
+++ b/wrapcheck/testdata/config_ignorePackageGlobs_interface/.wrapcheck.yaml
@@ -1,0 +1,2 @@
+ignorePackageGlobs:
+  - encoding/*

--- a/wrapcheck/testdata/config_ignorePackageGlobs_interface/main.go
+++ b/wrapcheck/testdata/config_ignorePackageGlobs_interface/main.go
@@ -1,0 +1,19 @@
+package main
+
+import (
+	"encoding/json"
+	"time"
+)
+
+func main() {
+	do(&time.Time{})
+}
+
+func do(fn json.Unmarshaler) error {
+	err := fn.UnmarshalJSON([]byte{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}

--- a/wrapcheck/wrapcheck.go
+++ b/wrapcheck/wrapcheck.go
@@ -234,7 +234,16 @@ func run(cfg WrapcheckConfig) func(*analysis.Pass) (interface{}, error) {
 
 // Report unwrapped takes a call expression and an identifier and reports
 // if the call is unwrapped.
-func reportUnwrapped(pass *analysis.Pass, call *ast.CallExpr, tokenPos token.Pos, cfg WrapcheckConfig, regexpsSig []*regexp.Regexp, regexpsInter []*regexp.Regexp, pkgGlobs []glob.Glob) {
+func reportUnwrapped(
+	pass *analysis.Pass,
+	call *ast.CallExpr,
+	tokenPos token.Pos,
+	cfg WrapcheckConfig,
+	regexpsSig []*regexp.Regexp,
+	regexpsInter []*regexp.Regexp,
+	pkgGlobs []glob.Glob,
+) {
+
 	sel, ok := call.Fun.(*ast.SelectorExpr)
 	if !ok {
 		return
@@ -242,7 +251,6 @@ func reportUnwrapped(pass *analysis.Pass, call *ast.CallExpr, tokenPos token.Pos
 
 	// Check for ignored signatures
 	fnSig := pass.TypesInfo.ObjectOf(sel.Sel).String()
-
 	if contains(cfg.IgnoreSigs, fnSig) {
 		return
 	} else if containsMatch(regexpsSig, fnSig) {
@@ -253,9 +261,9 @@ func reportUnwrapped(pass *analysis.Pass, call *ast.CallExpr, tokenPos token.Pos
 	// errors returned from interface types should be wrapped, unless ignored
 	// as per `ignoreInterfaceRegexps`
 	if isInterface(pass, sel) {
+		pkgPath := pass.TypesInfo.ObjectOf(sel.Sel).Pkg().Path()
 		name := types.TypeString(pass.TypesInfo.TypeOf(sel.X), func(p *types.Package) string { return p.Name() })
-		if containsMatch(regexpsInter, name) {
-		} else {
+		if !containsMatch(regexpsInter, name) && !containsMatchGlob(pkgGlobs, pkgPath) {
 			pass.Reportf(tokenPos, "error returned from interface method should be wrapped: sig: %s", fnSig)
 			return
 		}
@@ -318,6 +326,7 @@ func prevErrAssign(pass *analysis.Pass, file *ast.File, returnIdent *ast.Ident) 
 				if !isError(pass.TypesInfo.TypeOf(expr)) {
 					continue
 				}
+
 				if assIdent, ok := expr.(*ast.Ident); ok {
 					if assIdent.Obj == nil || returnIdent.Obj == nil {
 						// If we can't find the Obj for one of the identifiers, just skip
@@ -341,6 +350,7 @@ func prevErrAssign(pass *analysis.Pass, file *ast.File, returnIdent *ast.Ident) 
 		if ass.Pos() > returnIdent.Pos() {
 			break
 		}
+
 		mostRecentAssign = ass
 	}
 
@@ -373,6 +383,7 @@ func containsMatchGlob(globs []glob.Glob, el string) bool {
 			return true
 		}
 	}
+
 	return false
 }
 

